restructuredtext_lint/lint.py

```diff
@@ -80,9 +75,24 @@ def error_collector(data):
     return errors
 
 
-def lint_file(filepath, encoding=None):
+def lint(content, filepath=None, ignore_sphinx=True):
```

Comment:
I think we are missing each other. I mean making this generic and not specific to ignoring Sphinx. It should be able to ignore any given roles/directives via passed in parameters.

For making it easy for Sphinx, we can provide constants for users to pass in:

```python
def lint(content, filepath=None, custom_directives=None, custom_roles=None):
  with rst_lint_overrides(custom_directives, custom_roles):
    return _lint(content, filepath=filepath)

SPHINX_IGNORE_DIRECTIVES = [
  # Whatever a directive looks like
]

SPHINX_IGNORE_ROLES = [
  # Whatever a directive looks like
]
```